go/vt/proto/replicationdata/replicationdata.pb.go

```diff
@@ -24,7 +24,9 @@ const _ = proto.ProtoPackageIsVersion3 // please upgrade the proto package
 // Status is the replication status for MySQL (returned by 'show slave status'
 // and parsed into a Position and fields).
 type Status struct {
-	Position             string   `protobuf:"bytes,1,opt,name=position,proto3" json:"position,omitempty"`
+	Position string `protobuf:"bytes,1,opt,name=position,proto3" json:"position,omitempty"`
+	// RelayLogPosition will hold the Position the relay log is at. This is NOT supported in MariaDB, and will be empty for that flavor.
+	RelayLogPosition     string   `protobuf:"bytes,1,opt,name=position,proto3" json:"relayLogPosition,omitempty"`
```

Comment:
I see why this made the test fail. Protobuf uses numeric field IDs on the wire, and you've copied the field ID (1) of an existing field here.

In general, the way we should update these generated `.pb.go` files is to edit the corresponding `.proto` file (in this case `proto/replicationdata.proto`) and then run `make proto`.

You can add the new field underneath `position` since they go together semantically, but make sure to give the new field the next available ID (in this case 8).

Reply:
I'm curious why I wouldn't give it ID 2 and +1 the other ids so they are all in numerical order as well as visual order?

Reply:
Done as you suggested, but still curious to hear an answer to this at some point so I'll leave it un-resolved.

Reply:
For backwards compatibility you always add new fields at the end. That way if two processes are talking to one another that are off by a version, things still work. The new field will be received as nil/empty by the process that knows about it and invisible to the process that doesn't. For this reason you also have a 2-step process to delete fields from a proto. You first deprecate them (using `reserved`) and delete them in the next release.

Reply:
Thanks for the info!